### PR TITLE
eslint_d: repackage using buildNpmPackage

### PR DESCRIPTION
**Pull request by ehllie**
###### Description of changes
Repackages `nodePackages.eslint_d` to be built using `buildNpmPackage` as per the suggestion in #229475

###### Things done
Moved `eslint_d` to `pkgs/development/tools`, removed it from `node-packages.json` and included it in `aliases.nix`.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ehllie** [comment]
I've noticed the aliases have a comment next to them indicating when they were added. Are they supposed to be the time the change gets merged into nixpkgs, in which case I can't tell make that comment when creating the PR, or just when the PR gets submitted?

**Comment by SuperSandro2000** [comment]
The date is the time of writing. It is not important that it is exact to the date as it only gets used later to know if the alias entry is 10 months or 10 days old. With the next stable cycle it should be converted to a throw and with the one after that be removed.

File: pkgs/development/node-packages/aliases.nix [hunk 1]
```diff
@@ -44,4 +44,5 @@ mapAliases {
   "@nestjs/cli" = pkgs.nest-cli; # Added 2023-05-06
   manta = pkgs.node-manta; # Added 2023-05-06
   triton = pkgs.triton; # Added 2023-05-06
+  eslint_d = pkgs.eslint_d;
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/aliases.nix:47]:
```suggestion
  eslint_d = pkgs.eslint_d; # Added 2023-05-26
```
also we should sort the entry :)

File: pkgs/development/tools/eslint_d/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildNpmPackage, fetchFromGitHub }:
+
+buildNpmPackage rec {
+  pname = "eslint_d";
+  version = "12.2.1";
+
+  src = fetchFromGitHub {
+    owner = "mantoni";
+    repo = "${pname}.js";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/eslint_d/default.nix:9]:
```suggestion
    repo = "eslint_d.js";
```

**Pull request merged**